scripts/check-events

```diff
@@ -198,25 +206,35 @@ class Event
     # Validate <Namespace>EventsDropped events
     if is_events_dropped_event && !@skip_dropped_events
 
-      # Name check
-      append('EventsDropped events MUST be named "___EventsDropped".') unless @name.end_with? 'EventsDropped'
-      # Outputs an error log
-      handle.logs_should_have('error')
-      # Metric check
-      counters_must_include_exclude_tags(METRIC_NAME_EVENTS_DROPPED, ['intentional'], ['reason', 'count'])
+      # Don't run the checks on event structs which themselves emit ComponentEventsDropped,
+      # as the ComponentEventsDropped event is already checked.
+      # Instead, verify that component_discarded_events_total is not being over-incremented.
+      if @emits_component_events_dropped
+        if @counters.include? METRIC_NAME_EVENTS_DROPPED
+          @reports.append("Event emitting ComponentEventsDropped should not also increment counter `#{METRIC_NAME_EVENTS_DROPPED}`")
+        end
+      else
 
-      # Make sure EventsDropped events contain the required parameters
-      handle.logs.each do |type, message, parameters|
-        if type == 'error'
-          ['intentional', 'reason'].each do |parameter|
-            unless parameters.include? parameter
-              @reports.append("Error log for EventsDropped event MUST include parameter \"#{parameter}\".")
+        # Name check
+        append('EventsDropped events MUST be named "___EventsDropped".') unless @name.end_with? 'EventsDropped'
+        # Outputs an error log
+        handle.logs_should_have('error')
```

Comment:
That seems to be incorrect in regard to https://github.com/vectordotdev/vector/issues/14261. Should check for `debug` if `intentional` is `true`